api/debug/flags.go

```diff
@@ -136,37 +186,91 @@ func CreateLogDir(logDir string) {
+	var ostream log.Handler
+	output := io.Writer(os.Stderr)
+	if ctx.GlobalBool(logjsonFlag.Name) {
+		ostream = log.StreamHandler(output, log.JsonFormat())
+	} else {
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) && os.Getenv("TERM") != "dumb")
+		if usecolor {
+			output = colorable.NewColorableStderr()
+		}
+		ostream = log.StreamHandler(output, log.TerminalFormat(usecolor))
+	}
+	glogger.SetHandler(ostream)
+
 	// logging
-	log.PrintOrigins(ctx.GlobalBool(debugFlag.Name))
+	log.PrintOrigins(getGlobalBoolByFlagWithLegacy(ctx, debugFlag, legacyDebugFlag, "(v1.12.0)"))
 	log.ChangeGlobalLogLevel(glogger, log.Lvl(ctx.GlobalInt(verbosityFlag.Name)))
 	glogger.Vmodule(ctx.GlobalString(vmoduleFlag.Name))
-	glogger.BacktraceAt(ctx.GlobalString(backtraceAtFlag.Name))
+
+	glogger.BacktraceAt(getGlobalStringByFlagWithLegacy(ctx, backtraceAtFlag, legacyBacktraceAtFlag, "(v1.12.0)"))
 	log.Root().SetHandler(glogger)
 
 	// profiling, tracing
-	runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
-	Handler.SetBlockProfileRate(ctx.GlobalInt(blockprofilerateFlag.Name))
+	runtime.MemProfileRate = getGlobalIntByFlagWithLegacy(ctx, memprofilerateFlag, legacyMemprofilerateFlag, "(v1.12.0)")
+
+	Handler.SetBlockProfileRate(getGlobalIntByFlagWithLegacy(ctx, blockprofilerateFlag, legacyBlockprofilerateFlag, "(v1.12.0)"))
+
 	if traceFile := ctx.GlobalString(traceFlag.Name); traceFile != "" {
 		if err := Handler.StartGoTrace(traceFile); err != nil {
 			return err
 		}
 	}
-	if cpuFile := ctx.GlobalString(cpuprofileFlag.Name); cpuFile != "" {
-		if err := Handler.StartCPUProfile(cpuFile); err != nil {
+	if cpuProfile := getGlobalStringByFlagWithLegacy(ctx, cpuprofileFlag, legacyCpuprofileFlag, "(v1.12.0)"); cpuProfile != "" {
+		if err := Handler.StartCPUProfile(cpuProfile); err != nil {
 			return err
 		}
 	}
-	Handler.memFile = ctx.GlobalString(memprofileFlag.Name)
+
+	Handler.memFile = getGlobalStringByFlagWithLegacy(ctx, memprofileFlag, legacyMemprofileFlag, "(v1.12.0)")
 
 	// pprof server
 	if ctx.GlobalBool(pprofFlag.Name) {
-		addr := ctx.GlobalString(pprofAddrFlag.Name)
-		port := ctx.GlobalInt(pprofPortFlag.Name)
-		Handler.StartPProf(&addr, &port)
+		addr := getGlobalStringByFlagWithLegacy(ctx, pprofAddrFlag, legacyPprofAddrFlag, "(v1.12.0)")
+		pprofPort := getGlobalIntByFlagWithLegacy(ctx, pprofPortFlag, legacyPprofPortFlag, "(v1.12.0)")
+		Handler.StartPProf(&addr, &pprofPort)
 	}
 	return nil
 }
 
+func getGlobalStringByFlagWithLegacy(ctx *cli.Context, flag cli.StringFlag, legacyFlag cli.StringFlag, deprecateVersion string) string {
+	if ctx.GlobalIsSet(flag.Name) {
+		if globalString := ctx.GlobalString(flag.Name); globalString != "" {
+			return globalString
+		}
+	}
+	if ctx.GlobalIsSet(legacyFlag.Name) {
+		if globalString := ctx.GlobalString(legacyFlag.Name); globalString != "" {
+			logger.Warn("The flag" + legacyFlag.Name + " is deprecated and will be removed in the future " + deprecateVersion + ", please use " + flag.Name)
```

Comment:
The warning msg has typos. Could you check it and revise those warning msgs?

Reply:
Oh, I revised it. Thank you.